### PR TITLE
Bug 1168324 - Migrate test_sms_contact_match.py to Gij

### PR DESCRIPTION
**Pull request by JohanLorenzo**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b6567fb0e35d62ef2366c45230b98e3e03832fd0

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=22edeb3bd1c7cfdee4168db70fae8574edbd2e3c

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bd579ff5fd4f3a24764916d39311f353c234c812

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=acf979d95f0a3fdea187eb8d3928a00d77b5f355

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ef4f4a1370b983446b1ff02ac65be8053715d631

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=459dfb204d12e85e01c08cdf8f9ffeb9ed5b9fab

File: apps/sms/test/marionette/composer_test.js [hunk 1]
```diff
@@ -257,38 +258,73 @@ marionette('Messages Composer', function() {
   });
 
 
-  suite('Invalid recipients', function() {
+  suite('Recipients list', function() {
     var newMessage;
```
Comment (azasypkin) [inline apps/sms/test/marionette/composer_test.js]: nit: 
```
var newMessage, storage;

var contact = {
```

File: apps/sms/test/marionette/composer_test.js [hunk 2]
```diff
@@ -257,38 +258,73 @@ marionette('Messages Composer', function() {
...
+      familyName: ['Contact'],
+      tel: [{
+        value: '5551234567',
+        type: 'Mobile'
+      }]
+    };
 
     setup(function() {
-      messagesApp.launch();
+      var MOCKS = [
```
Comment (azasypkin) [inline apps/sms/test/marionette/composer_test.js]: nit: move ```MOCKS``` in the parent scope, where the rest of "const" variables (```newMessage storage contact```) are defined

File: apps/sms/test/marionette/composer_test.js [hunk 3]
```diff
@@ -257,38 +258,73 @@ marionette('Messages Composer', function() {
...
+      storage = Storage.create(client);
+
+      MOCKS.forEach(function(mock) {
+        client.contentScript.inject(__dirname + mock);
+      });
 
+      messagesApp.launch();
+      storage.setMessagesStorage();
+      storage.setContactsStorage([contact]);
       var inbox = new InboxView(client);
```
Comment (azasypkin) [inline apps/sms/test/marionette/composer_test.js]: nit: new line before ```var inbox```

File: apps/sms/test/marionette/composer_test.js [hunk 4]
```diff
@@ -257,38 +258,73 @@ marionette('Messages Composer', function() {
...
       newMessage = inbox.createNewMessage();
     });
 
-    suite('the recipients list', function() {
-      test('should display that a non existing contact is invalid', function() {
-        newMessage.addNewRecipient('non_exisiting_contact');
-        assert.isTrue(newMessage.containsInvalidRecipients());
+    teardown(function() {
```
Comment (azasypkin) [inline apps/sms/test/marionette/composer_test.js]: mmm why do we need this weird ```teardown```?

File: apps/sms/test/marionette/composer_test.js [hunk 5]
```diff
@@ -257,38 +258,73 @@ marionette('Messages Composer', function() {
...
+      storage.setContactsStorage();
+    });
+
```
Comment (azasypkin) [inline apps/sms/test/marionette/composer_test.js]: nit: redundant new line

File: apps/sms/test/marionette/composer_test.js [hunk 6]
```diff
@@ -257,38 +258,73 @@ marionette('Messages Composer', function() {
...
+
+    suite('The recipients', function() {
```
Comment (azasypkin) [inline apps/sms/test/marionette/composer_test.js]: Can we have better names for the suits here (can't understand why we need ```The recipients``` suite specifically)? Now I see something not really valuable or clear :)
```
Recipients list
	The recipients
    	✓ should match an existing contact (2048ms)
    	the recipients list
        	✓ should display that a non existing contact is invalid (2131ms)
        	✓ should allow to correct an invalid contact (6286ms)
        Content composer
        	✓ should not enable send button if the contact is invalid (2699ms)
```

File: apps/sms/test/marionette/lib/views/new-message/view.js [hunk 7]
```diff
@@ -22,6 +22,13 @@ NewMessageView.prototype = {
     });
   },
 
+  get recipientsPhoneNumbers() {
+    var recipients = this.accessors.recipients;
+    return recipients.map(function(recipient) {
```
Comment (azasypkin) [inline apps/sms/test/marionette/lib/views/new-message/view.js]: nit: no need in additional variable, just use ```this.accessors.recipients.map``` when it fits into 80 chars.

File: apps/sms/test/marionette/composer_test.js (hunk 6 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/composer_test.js]: My bad, wrong merge during the cherry pick. Here's what we get now:
>     Recipients
      ✓ should match an existing contact
      Invalid recipients
        Recipients list
          ✓ should display that a non existing contact is invalid (1859ms)
          ✓ should allow to correct an invalid contact (4490ms)
        Content composer
          ✓ should not enable send button if the contact is invalid (2553ms)

File: apps/sms/test/marionette/composer_test.js (hunk 5 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/composer_test.js]: Done.

File: apps/sms/test/marionette/composer_test.js (hunk 4 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/composer_test.js]: We don't. That's a leftover.

File: apps/sms/test/marionette/composer_test.js (hunk 3 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/composer_test.js]: Done

File: apps/sms/test/marionette/composer_test.js (hunk 2 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/composer_test.js]: Done

File: apps/sms/test/marionette/composer_test.js (hunk 1 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/composer_test.js]: Done

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b659b0a990673ca5afece76b88289e727ed063dc

**Pull request merged**